packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/features/resolution/ResolutionFeature.java

```diff
@@ -164,13 +184,72 @@ public static CamcorderProfile getBestAvailableCamcorderProfileForResolutionPres
     }
   }
 
-  private void configureResolution(ResolutionPreset resolutionPreset, int cameraId) {
+  @TargetApi(Build.VERSION_CODES.S)
+  public static EncoderProfiles getBestAvailableCamcorderProfileForResolutionPresetOn31(
+      int cameraId, ResolutionPreset preset) {
+    if (cameraId < 0) {
+      throw new AssertionError(
+          "getBestAvailableCamcorderProfileForResolutionPreset can only be used with valid (>=0) camera identifiers.");
+    }
+
+    String cameraIdString = Integer.toString(cameraId);
+
+    switch (preset) {
+        // All of these cases deliberately fall through to get the best available profile.
+      case max:
+        if (CamcorderProfile.hasProfile(cameraId, CamcorderProfile.QUALITY_HIGH)) {
+          return CamcorderProfile.getAll(cameraIdString, CamcorderProfile.QUALITY_HIGH);
+        }
+      case ultraHigh:
+        if (CamcorderProfile.hasProfile(cameraId, CamcorderProfile.QUALITY_2160P)) {
+          return CamcorderProfile.getAll(cameraIdString, CamcorderProfile.QUALITY_2160P);
+        }
+      case veryHigh:
+        if (CamcorderProfile.hasProfile(cameraId, CamcorderProfile.QUALITY_1080P)) {
+          return CamcorderProfile.getAll(cameraIdString, CamcorderProfile.QUALITY_1080P);
+        }
+      case high:
+        if (CamcorderProfile.hasProfile(cameraId, CamcorderProfile.QUALITY_720P)) {
+          return CamcorderProfile.getAll(cameraIdString, CamcorderProfile.QUALITY_720P);
+        }
+      case medium:
+        if (CamcorderProfile.hasProfile(cameraId, CamcorderProfile.QUALITY_480P)) {
+          return CamcorderProfile.getAll(cameraIdString, CamcorderProfile.QUALITY_480P);
+        }
+      case low:
+        if (CamcorderProfile.hasProfile(cameraId, CamcorderProfile.QUALITY_QVGA)) {
+          return CamcorderProfile.getAll(cameraIdString, CamcorderProfile.QUALITY_QVGA);
+        }
+      default:
+        if (CamcorderProfile.hasProfile(cameraId, CamcorderProfile.QUALITY_LOW)) {
+          return CamcorderProfile.getAll(cameraIdString, CamcorderProfile.QUALITY_LOW);
+        }
+
+        throw new IllegalArgumentException(
+            "No capture session available for current capture session.");
+    }
+  }
+
+  @SuppressWarnings("deprecation")
```

Comment:
This was marked as resolved, but is still present; is that a mistake, or is there a reason we need this? If the latter, please explain in a reply so that we have a record of it for people trying to understand this code later.